pilot/pkg/networking/core/v1alpha3/loadbalancer/loadbalancer_test.go

```diff
@@ -24,21 +24,67 @@ func TestApplyLocalitySetting(t *testing.T) {
 	}
 
 	t.Run("Distribute", func(t *testing.T) {
-		g := NewGomegaWithT(t)
-		env := buildEnvForClustersWithDistribute()
-		cluster := buildFakeCluster()
-		ApplyLocalityLBSetting(locality, cluster.LoadAssignment, env.Mesh.LocalityLbSetting)
-
-		for _, localityEndpoint := range cluster.LoadAssignment.Endpoints {
-			if util.LocalityMatch(localityEndpoint.Locality, "region1/zone1/subzone1") {
-				g.Expect(localityEndpoint.LoadBalancingWeight.GetValue()).To(Equal(uint32(90)))
-				continue
-			}
-			if util.LocalityMatch(localityEndpoint.Locality, "region1/zone1") {
-				g.Expect(localityEndpoint.LoadBalancingWeight.GetValue()).To(Equal(uint32(5)))
-				continue
-			}
-			g.Expect(localityEndpoint.LbEndpoints).To(BeNil())
+		tests := []struct {
+			name       string
+			distribute []*meshconfig.LocalityLoadBalancerSetting_Distribute
+			expected   []int
+		}{
+			{
+				name: "distribution between subzones",
+				distribute: []*meshconfig.LocalityLoadBalancerSetting_Distribute{
+					{
+						From: "region1/zone1/subzone1",
+						To: map[string]uint32{
+							"region1/zone1/subzone1": 80,
+							"region1/zone1/subzone2": 15,
+							"region1/zone1/subzone3": 5,
+						},
+					},
+				},
+				expected: []int{40, 40, 15, 5, 0, 0, 0},
+			},
+			{
+				name: "distribute with overlapping to",
+				distribute: []*meshconfig.LocalityLoadBalancerSetting_Distribute{
+					{
+						From: "region1/zone1/subzone1",
+						To: map[string]uint32{
+							"region1/zone1/subzone1": 50,
+							"region1/zone1/*":        40,
```

Comment:
You may misunderstand： Here "region1/zone1/subzone1" and "region1/zone1/*" are overlapping

Previously I mean: if we only have  "region1/zone1/*", then it matches "region1/zone1/subzone1" "region1/zone1/subzone2" and so on.

Reply:
Back to the test, this is not a valid case